modules/storages/spec/support/shared_examples_for_storage_providers/files_info_query_examples.rb

```diff
@@ -0,0 +1,59 @@
+# frozen_string_literal: true
+
+#-- copyright
+# OpenProject is an open source project management software.
+# Copyright (C) the OpenProject GmbH
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See COPYRIGHT and LICENSE files for more details.
+#++
+
+RSpec.shared_examples_for "adapter files_info_query: basic query setup" do
+  it "is registered as queries.files_info" do
+    expect(Storages::Adapters::Registry
+             .resolve("#{storage}.queries.files_info")).to eq(described_class)
+  end
+
+  it "responds to #call with correct parameters" do
+    expect(described_class).to respond_to(:call)
+
+    method = described_class.method(:call)
+    expect(method.parameters).to contain_exactly(%i[keyreq storage],
+                                                 %i[keyreq auth_strategy],
+                                                 %i[keyreq input_data])
+  end
+end
+
+RSpec.shared_examples_for "adapter files_info_query: successful list response" do
+  it "returns an array of file info objects" do
+    result = described_class.call(storage:, auth_strategy:, input_data:)
+
+    expect(result).to be_success
+
+    response = result.value!
+    expect(response).to be_an(Array)
+    response.each { expect(it).to be_a(Storages::Adapters::Results::StorageFileInfo) }
```

Comment:
:yellow_circle: I think this is supported by RSpec as

```suggestion
    expect(response).to all(be_a(Storages::Adapters::Results::StorageFileInfo))
```